leaflet.filelayer.js

```diff
@@ -82,13 +88,81 @@ var FileLoader = L.Class.extend({
         }
         var geojson = toGeoJSON[format](content);
         return this._loadGeoJSON(geojson);
-    }
+    },
+
+    /*
+     *  Added from 4535992.
+     */
+    _convertCsvToGeoJSON: function(content){
+        try {
+            if(!this.options.firstLineTitles){
+                throw new Error('The file CSV must have the Headers');
+            }
+            var geojson;
+            /*
+             * for this function i used the Papa Parser of mholt
+             * (https://github.com/mholt/PapaParse)
+             */
+            geojson = Papa.parse(content,{header: this.options.firstLineTitles}); //convert csv to json.
+            geojson = this._addFeatureToJson(geojson);
+            return this._loadGeoJSON(geojson);
+        }catch(e){alert(e.message);}
+    },
+
+    _addFeatureToJson: function(papaJson){
+        papaJson["type"]="FeatureCollection";
+        papaJson["features"]=[];
+        var title = papaJson.meta.fields;
+        for (var num_linea = 0; num_linea < papaJson.data.length - 1; num_linea++) { //  var depth = papaJson.data.length - 1;
+            var obj = papaJson.data[num_linea]; //single element of papa parse json object
+            var fields = title.toString().trim().split(",");
+            var lng = parseFloat(obj[this.options.longitudeTitle]);
+            var lat = parseFloat(obj[this.options.latitudeTitle]);
+            if (fields.length+"=="+title.length && lng<180 && lng>-180 && lat<90 && lat>-90) {
+                var feature = {};
+                feature["type"]="Feature";
+                feature["geometry"]={};
+                feature["properties"]={};
+                feature["geometry"]["type"]="Point";
+                feature["geometry"]["coordinates"]=[lng,lat];
```

Comment:
Could be done in one line `feature["geometry"] = {"type": "Point", "coordinates": [lng, lat]}`